File: solutions/java/sieve/1/src/main/java/Sieve.java
import java.util.ArrayList;
import java.util.List;

class Sieve {
    List<Integer> primes = new ArrayList<Integer>();

    Sieve(int maxPrime) {

        if (maxPrime == 1) return;
        primes.add(2);
        for (int i = 3; i < maxPrime; i+=2) primes.add(i);
        for (int j = 1; j < primes.size(); j++) {
            int k = j+1;
            while(k < primes.size()){
                //System.out.println("j="+j+",k="+k);
                if (primes.get(k) % primes.get(j) == 0) primes.remove(k);
                else k++;
            }
        }
    }

    List<Integer> getPrimes() {
        return(primes);
    }
}

File: solutions/cpp/sieve/1/sieve.cpp
#include "sieve.h"
using namespace std;

namespace sieve {
    vector<int> primes(int n) {
        vector<int> p = {};
        if (n==1) return p;
        for (int i=2; i<=n; i++) {
            int j;
            for (j=0; j<(int)p.size();j++) {
                if (i % p[j]==0) break;
            }
            if (j==(int)p.size()) p.push_back(i);
        }
        return p;
    }
}  // namespace sieve
